app/src/androidTest/java/com/android/sample/screen/mainscreen/CalendarScreenTest.kt

```diff
@@ -0,0 +1,127 @@
+        listOf(
+            BodyWeightWorkout(
+                "2",
+                "NightSes",
+                "Hold for 60 seconds",
+                false,
+                date = LocalDateTime.of(2024, 11, 1, 2, 1)))
+    val yogaWorkouts: List<YogaWorkout> = listOf()
+
+    `when`(bodyWeightRepo.getDocuments(any(), any())).then {
+      it.getArgument<(List<BodyWeightWorkout>) -> Unit>(0)(bodyWeightWorkouts)
+    }
+
+    `when`(bodyWeightRepo.deleteDocument(any(), any(), any())).then {
+      it.getArgument<(List<BodyWeightWorkout>) -> Unit>(0)(emptyList())
+    }
+
+    `when`(yogaRepo.getDocuments(any(), any())).then {
+      it.getArgument<(List<YogaWorkout>) -> Unit>(0)(yogaWorkouts)
+    }
+
+    bodyWeightViewModel = WorkoutViewModel(bodyWeightRepo)
+    yogaViewModel = WorkoutViewModel(yogaRepo)
+
+    // Mock the NavigationActions
+    navigationActions = mock(NavigationActions::class.java)
+  }
+
+  @Test
+  fun displayAllComponents() {
+    composeTestRule.setContent {
+      CalendarScreen(navigationActions, bodyWeightViewModel, yogaViewModel)
+    }
+    bodyWeightViewModel.getWorkouts()
+
+    // Check if the top bar, legend, and lazy column are displayed
+    composeTestRule.onNodeWithTag("workoutItem").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("TopBar").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("legendYoga").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("legendBodyweight").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("lazyColumn").assertIsDisplayed()
+  }
+
+  @Test
+  fun testNavigationOnBack() {
+    composeTestRule.setContent {
+      CalendarScreen(
+          navigationActions = navigationActions,
+          bodyworkoutViewModel = bodyWeightViewModel,
+          yogaworkoutViewModel = yogaViewModel)
+    }
+
+    composeTestRule.onNodeWithTag("ArrowBackButton").performClick()
+
+    verify(navigationActions).goBack()
+  }
+
+  @Test
+  fun testWorkoutClickShowsDialog() {
+
+    composeTestRule.setContent {
+      CalendarScreen(navigationActions, bodyWeightViewModel, yogaViewModel)
+    }
+
+    bodyWeightViewModel.getWorkouts()
+
+    // Simulate a workout click and ensure the dialog shows up
+    composeTestRule.onNodeWithTag("workoutItem").performClick()
+    composeTestRule.onNodeWithTag("alertDialog").assertIsDisplayed()
+
+    // Simulate click on the 'Edit' button
+    composeTestRule.onNodeWithTag("editButton").performClick()
+
+    // Verify that the dialog is dismissed
+    composeTestRule.onNodeWithTag("alertDialog").assertDoesNotExist()
+  }
+
+  @Test
+  fun testDisplayMoreThan3Days() {
+    composeTestRule.setContent {
+      CalendarScreen(navigationActions, bodyWeightViewModel, yogaViewModel)
+    }
+
+    // Find all nodes with the testTag "daySection"
+    val dayNodes = composeTestRule.onAllNodesWithTag("daySection", useUnmergedTree = true)
+
+    // Assert that there are 7 day sections displayed
+    assert(3 < dayNodes.fetchSemanticsNodes().size)
+  }
```

Comment:
Why 3?

Reply:
This comes from that the size of the CI screen is small. 

But i think it is a good check to see if multiple days appears.

(The typo is fixed)